rtrlib/rtr/rtr.c

```diff
@@ -52,6 +53,127 @@ static int install_sig_handler()
     return sigaction(SIGUSR1, &sa, NULL);
 }
 
+int get_interval_mode()
+{
+    return interval_option;
+}
+
+void set_interval_mode(int option)
+{
+    const char* txt;
+    switch(option) {
+    case IGNORE_ANY:
+        interval_option = option;
+        txt = "IGNORE_ANY";
+        break;
+    case ACCEPT_ANY:
+        interval_option = option;
+        txt = "ACCEPT_ANY";
+        break;
+    case DEFAULT_MIN_MAX:
+        interval_option = option;
+        txt = "DEFAULT_MIN_MAX";
+        break;
+    case IGNORE_ON_FAILURE:
+        interval_option = option;
+        txt = "IGNORE_ON_FAILURE";
+        break;
+    default:
+        RTR_DBG1("Invalid interval mode. Mode remains unchanged.");
+        return;
+    }
+    RTR_DBG("Interval mode set to %s", txt);
+    /*
+    if(option == IGNORE_ANY ||
+       option == ACCEPT_ANY ||
+       option == DEFAULT_MIN_MAX ||
+       option == IGNORE_ON_FAILURE) {
+        interval_option = option;
+        RTR_DBG("Interval mode set to: %u", get_interval_mode());
+    } else {
+        RTR_DBG1("Invalid interval mode. Mode remains unchanged.");
+    }
+     */
+}
+
+int rtr_check_interval_range(uint32_t interval, int16_t minimum, int32_t maximum)
+{
+    if(interval < minimum) {
+        return BELOW_INTERVAL_RANGE;
+    } else if (interval > maximum) {
+        return ABOVE_INTERVAL_RANGE;
+    }
+
+    return INSIDE_INTERVAL_RANGE;
+}
+
+void apply_interval_value(struct rtr_socket *rtr_socket,
+                          uint32_t interval,
+                          enum interval_type type)
+{
+    if (type == EXPIRATION) {
+        rtr_socket->expire_interval = interval;
+    } else if (type == REFRESH) {
+        rtr_socket->refresh_interval = interval;
+    } else if (type == RETRY) {
+        rtr_socket->retry_interval = interval;
+    }
+}
+
+int rtr_check_interval_option(struct rtr_socket *rtr_socket,
+                                     int interval_mode,
+                                     uint32_t interval,
+                                     enum interval_type type) {
+    uint16_t minimum;
+    uint32_t maximum;
+
+    int interv_retval;
+
+    switch (type) {
+    case EXPIRATION:
+        minimum = RTR_EXPIRATION_MIN;
+        maximum = RTR_EXPIRATION_MAX;
```

Comment:
To prevent confusion on my side: should minimum and maximum be initialized before the switch statement like this?
`uint16_t minimum = 0;`
`uint32_t maximum = 0;`